src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -495,7 +495,7 @@ private Conversion ClassifyStandardImplicitConversion(TypeSymbol source, TypeSym
 
             if (HasBoxingConversion(source, destination, ref useSiteDiagnostics))
             {
-                return Conversion.Boxing;
+                return source.Kind == SymbolKind.TypeParameter ? Conversion.TypeParameterBoxing : Conversion.ValueTypeBoxing;
```

Comment:
I assume this is the intended classification if the type parameter is constrained to `struct`.